sale_coupon_multi_use/models/sale_coupon.py

```diff
@@ -57,9 +60,22 @@ def _get_related_sale_order_line(self, sale_order):
 
     def _prepare_consumption_line(self, sale_order_line):
         self.ensure_one()
+        program = self.program_id
+        from_currency = sale_order_line.order_id.pricelist_id.currency_id
+        to_currency = program.currency_id
+        amount = from_currency._convert(
+            abs(sale_order_line.price_total),
+            to_currency,
+            # program company is not mandatory, so fallback to user
+            # company.
+            program.company_id or self.env.user.company_id,
+            # Symmetrical -> program discount is also converted to SO
+            # amount using today.
+            fields.Date.today(),
+        )
         return {
             "coupon_id": self.id,
-            "amount": abs(sale_order_line.price_total),
+            "amount": amount,
```

Comment:
To move in your `sale_coupon_multi_currency` module: it's not linked to multi use.